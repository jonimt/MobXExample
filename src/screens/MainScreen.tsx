import React, {FC} from 'react';
import {SafeAreaView, StyleSheet} from 'react-native';
import {Header} from '../components/Header';
import {TodoFooter} from '../components/TodoFooter';
import {TodoList} from '../components/TodoList';
import {todoStore} from '../store/TodoStore';

export const MainScreen: FC = () => {
  return (
    <SafeAreaView style={styles.container}>
      <Header />
      <TodoList todos={todoStore.todos} />
      <TodoFooter />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
