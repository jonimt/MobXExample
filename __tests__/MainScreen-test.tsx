import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import {MainScreen} from '../src/screens/MainScreen';

describe('MainScreen testing', () => {
  // Snapshot test
  it('renders correctly', () => {
    const tree = renderer.create(<MainScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
