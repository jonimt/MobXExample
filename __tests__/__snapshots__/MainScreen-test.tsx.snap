// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen testing renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "black",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
          "fontFamily": "HelveticaNeue-Medium",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      Todos
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "HelveticaNeue",
          "fontSize": 18,
          "position": "absolute",
          "right": 15,
        }
      }
    >
      0
      /
      0
    </Text>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
    style={
      Object {
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 15,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="todo"
        style={
          Object {
            "borderColor": "black",
            "borderRadius": 25,
            "borderWidth": 1,
            "height": 50,
            "paddingHorizontal": 8,
            "textAlign": "center",
            "width": "100%",
          }
        }
        value=""
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "pink",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginHorizontal": 15,
          "marginTop": 8,
        }
      }
      testID="AddTodoButton"
    >
      <Text>
        Add Todo
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
